src/inference/dev_api/performance_heuristics.hpp

```diff
@@ -48,7 +49,7 @@ static MemBandwidthPressure MemBandwidthPressureTolerance(
             std::strcmp("ConvolutionBackpropData", node_name)) {
             if (!std::strcmp("GRUSequence", node_name) || !std::strcmp("TensorIterator", node_name)) {
                 MemBandwidthPressure res;
-                res.max_mem_tolerance = MemBandwidthPressure::UNKNOWN;
+                res.max_mem_tolerance = MemBandwidthPressure::SPECIAL;
```

Comment:
This SPECIAL case looks like WA to disable existing math model. Any details why we need it at all?
Additional questions how it affects x86 behavior?

Reply:
@dmitry-gorokhov In mask branch, below two kinds of models will get same `UNKNOWN` result and set same model preferred threads per stream.

1. Model include `MatMul` or `Convolution` or `ConvolutionBackpropData`, and also include `GRUSequence` or `TensorIterator`
2. Model does not include `MatMul` or `Convolution` or `ConvolutionBackpropData`

However, in ARM server profiling, those two kinds of models prefer different threads per stream. So in this PR, the first kind of model will return `SPECIAL` and set different model preferred threads per stream.

Reply:
Thanks for details.
As I mentioned this is obvious WA that breaks underlaying math model. Besides that such solution affects behavior of GPU plugin: https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/src/plugin/plugin.cpp#L780. I cannot accept current solution.
Proper implemention should provide expliit handling for these operations:
1. TI is composite operations that contains a body. We need to recursivly call MemBandwidthPressureTolerance for the body get correct reults.
2. GRUSequence should contain own processing logic. I suppose it should be quite close to Matmul block (maybe we can even reuse Mamtul block at all)

Reply:
Got it. I reverted the WA of SPECIAL models and just keep tunning for modelPreferThreads based on current mem_bandwidth_pressure_tolerance design. mem_bandwidth_pressure_tolerance refactor will be handled in following PR.

@dmitry-gorokhov is it ok for current PR?

Reply:
Yep, I think this is reasonable